hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoTransferFeeResultCalculator.java

```diff
@@ -0,0 +1,269 @@
+// SPDX-License-Identifier: Apache-2.0
+package com.hedera.node.app.service.token.impl.handlers;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.AccountID;
+import com.hedera.hapi.node.base.HederaFunctionality;
+import com.hedera.hapi.node.base.SubType;
+import com.hedera.hapi.node.base.TokenID;
+import com.hedera.hapi.node.base.TransferList;
+import com.hedera.hapi.node.state.token.Account;
+import com.hedera.hapi.node.state.token.Token;
+import com.hedera.hapi.node.token.CryptoTransferTransactionBody;
+import com.hedera.hapi.node.transaction.TransactionBody;
+import com.hedera.node.app.spi.fees.AbstractSimpleFeeCalculator;
+import com.hedera.node.app.spi.fees.SimpleFeeCalculator;
+import com.hedera.node.config.data.EntitiesConfig;
+import com.hedera.node.config.data.HederaConfig;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import org.hiero.hapi.support.fees.Extra;
+
+/** Calculates CryptoTransfer fees. Estimates HBAR/token/NFT transfers, auto-creation, auto-associations, and custom fees. */
+public class CryptoTransferFeeResultCalculator extends AbstractSimpleFeeCalculator {
+
+    @Override
+    @NonNull
+    protected HederaFunctionality getFunctionality() {
+        return HederaFunctionality.CRYPTO_TRANSFER;
+    }
+
+    @Override
+    @NonNull
+    protected SubType getSubType(
+            @NonNull final TransactionBody body, @Nullable final SimpleFeeCalculator.TxContext context) {
+        return SubType.DEFAULT;
+    }
+
+    @Override
+    @NonNull
+    protected Map<Extra, Long> buildExtraParams(
```

Comment:
I also don't see why we need a map here. If we are doing this params calculation inside each calculator, we just need some local variables that count each extra and do some business logic